src/orchard/inspect.clj

```diff
@@ -356,18 +372,68 @@
         (render-ln))
     inspector))
 
+(defn- rendered-element-str
+  "Return the string representation of the rendered `element`."
+  [element]
+  (cond (string? element)
+        element
+        (and (seq? element) (= :value (first element)))
+        (second element)
+        (and (seq? element) (= :newline (first element)))
+        "\n"
+        :else
+        (throw (ex-info "Unsupported element" {:element element}))))
+
+(defn- render-map-key
+  "Render the key of a map."
+  [inspector key]
+  (render-value inspector key))
+
+(defn- long-map-key?
+  "Returns true of `s` is a long string, more than 20 character or
+  containing newlines."
+  [^String s]
+  (or (.contains s "\n") (> (count s) 20)))
+
+(defn- render-map-separator
+  "Render the map separator according to `rendered-key`. If
+  `rendered-key` is long or contains newlines the key and value will
+  be rendered on separate lines."
+  [{:keys [view-mode] :as inspector} rendered-key]
+  (if (and (= :pretty view-mode) (long-map-key? rendered-key))
```

Comment:
This and the few other predicates better be checked once and passed to these functions as an argument.

Reply:
I changed this now, and pass the result around.